elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -213,48 +198,66 @@ private Object fetchObject(RequestScope requestScope, Class recordType, Optional
+            String idField = dictionary.getIdFieldName(loadClass);
+            String entityTypeName = dictionary.getJsonAliasFor(loadClass);
 
             /* construct a new SQL like filter expression, eg: book.id IN [1,2] */
             filterExpression = Optional.of(new FilterPredicate(
                     new FilterPredicate.PathElement(
-                            recordType,
+                            loadClass,
                             entityTypeName,
                             idType,
                             idField),
                     Operator.IN,
                     new ArrayList<>(idList)));
 
-            return PersistentResource.loadRecords(recordType, requestScope, filterExpression);
+            return PersistentResource.loadRecords(loadClass, requestScope, filterExpression);
+        }
+    }
+
+    /**
+     * fetches a non-root level attribute or relationship
+     * @param dictionary entity dictionary
+     * @param parentResource parent resource
+     * @param fieldName field type
+     * @return attribute or relationship object
+     */
+    private Object fetchObject(EntityDictionary dictionary, PersistentResource parentResource,
+                               String fieldName) {
+        Class parentClass = parentResource.getResourceClass();
+
+        if(dictionary.isAttribute(parentClass, fieldName)) { /* fetch attribute properties */
+            return fetchAttribute(parentResource, fieldName);
+
+        } else if(dictionary.isRelation(parentClass, fieldName)){ /* fetch relationship properties */
+            return fetchRelationship(parentResource, fieldName);
+        } else {
+            throw new BadRequestException("Unrecognized object: " + fieldName + " for: " + parentClass.getName());
         }
     }
 
     /**
      * FETCH attributes of top level entity
      * @param parentResource parent object
-     * @param field Field type
-     * @return list of persistent resource objects
+     * @param fieldName Field type
+     * @return list of {@link PersistentResource} objects
      */
-    private Object fetchAttribute(PersistentResource parentResource, Field field) {
-        return parentResource.getAttribute(field.getName());
+    private Object fetchAttribute(PersistentResource parentResource, String fieldName) {
+        return parentResource.getAttribute(fieldName);
     }
 
     /**
      * Fetches a relationship for a top-level entity
-     * @param context environment encapsulating graphQL's request environment
+     * @param parentResource parent object
      * @return persistence resource object(s)
      */
-    private Object fetchRelationship(Environment context) {
-        String fieldName = context.field.getName();
-
-        Set<PersistentResource> relations = context.parentResource.getRelationCheckedFiltered(fieldName);
+    private Object fetchRelationship(PersistentResource parentResource, String fieldName) {
```

Comment:
It is odd for a function to return a Set in one instance and an single Object in another.